pkg/scheduler/extender_test.go

```diff
@@ -388,3 +390,85 @@ func TestIsInterested(t *testing.T) {
 		})
 	}
 }
+
+// TestRestoreNumPDBViolations test functions convertToNodeNameToMetaVictims and HTTPExtender.convertToNodeNameToVictims in HTTPExtender.ProcessPreemption
+// Ensure the NumPDBViolations of each node is returned from external server.
+func TestRestoreNumPDBViolations(t *testing.T) {
+	se := &HTTPExtender{}
+	nodes := []string{"node1", "node2"}
+	podsInNodeList := []*v1.Pod{
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod1", UID: "uid1"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod2", UID: "uid2"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod3", UID: "uid3"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "pod4", UID: "uid4"}},
+	}
+
+	nodeInfoList := make([]*framework.NodeInfo, 0, len(nodes))
+	nodeNameToVictims := make(map[string]*extenderv1.Victims)
+	result := extenderv1.ExtenderPreemptionResult{
+		NodeNameToMetaVictims: make(map[string]*extenderv1.MetaVictims),
+	}
+
+	// nodeNameToVictims{node1:{Pods: [pod1,pod3], NumPDBViolations: 1}, node2:{Pods: [pod2,pod4], NumPDBViolations: 2}}
+	// nodeInfoList[{node1:{Pods: pod1,pod3}}, {node2:{Pods: pod2,pod4}}]
+	// result.NodeNameToMetaVictims {node1:{MetaPod: [uid1,uid3], NumPDBViolations: 1}, node2:{MetaPod: [uid2,uid4], NumPDBViolations: 2}}
+	for i, nm := range nodes {
+		nodeInfo := framework.NewNodeInfo()
+		node := createNode(nm)
+		nodeInfo.SetNode(node)
+		nodeInfo.AddPod(podsInNodeList[i])
+		nodeInfo.AddPod(podsInNodeList[i+2])
+		nodeInfoList = append(nodeInfoList, nodeInfo)
+
+		victims := &extenderv1.Victims{NumPDBViolations: 1 + int64(i)}
+		victims.Pods = append(victims.Pods, podsInNodeList[i])
+		victims.Pods = append(victims.Pods, podsInNodeList[i+2])
+		nodeNameToVictims[nm] = victims
+
+		result.NodeNameToMetaVictims[nm] = &extenderv1.MetaVictims{NumPDBViolations: 1 + int64(i)}
+		result.NodeNameToMetaVictims[nm].Pods = append(result.NodeNameToMetaVictims[nm].Pods, &extenderv1.MetaPod{UID: string(podsInNodeList[i].UID)})
+		result.NodeNameToMetaVictims[nm].Pods = append(result.NodeNameToMetaVictims[nm].Pods, &extenderv1.MetaPod{UID: string(podsInNodeList[i+2].UID)})
+	}
+
+	for _, tc := range []struct {
+		label            string
+		nodeCacheCapable bool
```

Comment:
To cover `convertToNodeNameToMetaVictims()`, you can suppose this is always true.